builtin/credential/okta/backend.go

```diff
@@ -172,30 +172,33 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username, pas
 			break
 		}
 
-		factorAvailable := false
+		var selectedFactor, totpFactor, pushFactor *mfaFactor
 
-		var selectedFactor mfaFactor
-
-		// Okta push and totp are currently supported. If a totp passcode is provided during
-		// login and is supported, that will be the preferred method.
+		// Scan for available factors
 		for _, v := range result.Embedded.Factors {
+			v := v
 			if v.Provider != "OKTA" {
 				continue
 			}
 
-			if v.Type == mfaTOTPMethod && totp != "" {
-				selectedFactor = v
-				factorAvailable = true
-				break
-			}
-
-			if v.Type == mfaPushMethod {
-				selectedFactor = v
-				factorAvailable = true
+			switch v.Type {
+			case mfaTOTPMethod:
+				totpFactor = &v
+			case mfaPushMethod:
+				pushFactor = &v
 			}
 		}
 
-		if !factorAvailable {
+		// Okta push and totp are currently supported. If a totp passcode is provided during
+		// login and is supported, that will be the preferred method.
+		switch {
+		case totpFactor != nil && totp != "":
+			selectedFactor = totpFactor
+		case pushFactor != nil:
```

Comment:
This switch is pretty nice since it allows us to attempt doing push MFA if a totp is not provided and there are multiple factors enabled.